Updates to Sign-in to SPA with AuthJS (#4140)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md

1. [Test your app](#test-your-app) again and note that **Don't have an account?** with a **Sign up** link now appears for your app under the Sign-In Widget. Click the link to add a user. Enter your first name, last name, and email address, and then click **Sign up**. Click **Set up** to add a new password. (Click **Set up later** for any other authenticators.) You're now logged into the app with the new user's profile.

Based on other policy configurations, the self-service registration flow may be different or include other authenticators. See [Sign-in flows](https://help.okta.com/okta_help.htm?type=oie&id=ext-about-sign-in-flows).

> **Note:** All new users through the self-registration process receive a welcome email. This email activates user access to apps on your dev org and demonstrates ownership of the email authenticator.

## Add MFA with a mandatory second factor

By default, your dev org isn’t configured for multifactor authentication. Use the following steps to understand the policy configurations and set up this use case. This setup requires an end user to authenticate with a password and a phone authenticator.

> 1. Go to **Security > Authenticators** and ensure that the phone authenticator is available in the **Authenticators** list on the **Setup** tab.

Reviewer: ```suggestion
1. Go to **Security** > **Authenticators** and ensure that the phone authenticator is available in the **Authenticators** list on the **Setup** tab.
```

   1. If it isn't listed, click **Add Authenticator**, and then click **Add** in the **Phone** tile.
   {style="list-style-type:lower-alpha"}
   1. For **User can verify with**, select **SMS**.
   1. Set **This authenticator can be used for** to **Authentication and recovery**, and click **Add**.

> 1. Go to **Security > Authentication policies**, click **Add a policy**.

Reviewer: ```suggestion
1. Go to **Security** > **Authentication policies**, click **Add a policy**.
```

1. Add a name for the policy. For example, **Mandatory MFA**

1. In the **Catch-all Rule**, click **Edit** under the **Actions** dropdown menu.

1. For the option **User must authenticate with**, select **Password / IdP +  Another factor**. Ensure that the **Possession factor constraints are** doesn't have the **Exclude phone and email authenticators** checkbox selected. Click **Save**.

> 1. Go to **Applications > Applications** and select your app.

Reviewer: ```suggestion
1. Go to **Applications** > **Applications** and select your app.
```

> 1. Click the **Sign On** tab

Reviewer: ```suggestion
1. Click the **Sign On** tab.
```

1. Scroll down to the **User authentication** section and click **Edit**.

1. From the **Authentication policy** dropdown menu, select your new authentication policy, **Mandatory MFA**. Click **Save**.

> 1. Test the new configurations by logging into your app. If your test user doesn’t have a phone number enrolled, the user is prompted for the enrollment during the sign in. Any new users signing up also require the enrollment of the phone authenticator.

Reviewer: ```suggestion
1. Test the new configurations by signing into your app. If your test user doesn’t have a phone number enrolled, the user is prompted for the enrollment during the sign in. Any new users signing up also require the enrollment of the phone authenticator.
```

> After your users have enrolled in the phone authenticator, future sign-ins require both a password and SMS code to access your app.

Reviewer: ```suggestion
After your users have enrolled in the phone authenticator, future user sign-in flows require both a password and SMS code to access your app.
```

## Enable password recovery with email magic link

By default, the dev org is configured for a self-service password reset. Review the following steps to understand the policy configurations and to enable your sample app users to self-recover their password through an email magic link.

> 1. Go to **Security > Authenticators** and ensure the email authenticator is available in the **Authenticators** list on the **Setup** tab, and it's used for **Recovery**.

Reviewer: ```suggestion
1. Go to **Security** > **Authenticators** and ensure that the email authenticator is available in the **Authenticators** list on the **Setup** tab, and that it's used for **Recovery**.
```

   1. If it isn't listed, click **Add Authenticator**, and then click **Add** in the **Email** tile.
   {style="list-style-type:lower-alpha"}

>    1. Set **This authenticator can  be used for** to **Recovery**, and click **Add**.

Reviewer: ```suggestion
   1. Set **This authenticator can be used for** to **Recovery**, and click **Add**.
```

> 1. Go to **Security > Authenticators** and edit the **Password** authenticator, by clicking **Actions > Edit**.

Reviewer: ```suggestion
1. Go to **Security** > **Authenticators** and edit the **Password** authenticator by clicking **Actions** > **Edit**.
```

> 1. Scroll to the bottom of the **Default Policy**, and edit the **Default Rule**.

Reviewer: ```suggestion
1. Scroll to the bottom of the **Default Policy** and edit the **Default Rule**.
```

1. Ensure that **Password reset** is selected for the **Users can perform self-service** section.

1. Ensure that **Email** is selected for the **Users can initiate recovery with** section.

> 1. Click **Not required** for **additional verification is** question, and then click **Update rule**.

Reviewer: ```suggestion
1. Click **Not required** for the **additional verification is** question, and then click **Update rule**.
```

> 1. If you previously set the sign-on policy for your app as Mandatory MFA, go to **Applications > Applications** and select your app. Click the **Sign On** tab and scroll down to the **User authentication** section and click **Edit**. From the **Authentication policy** dropdown menu, select **One factor access** and click **Save**.

Reviewer: ```suggestion
1. If you previously set the sign-on policy for your app as Mandatory MFA, go to **Applications** > **Applications** and select your app. Click the **Sign On** tab and scroll down to the **User authentication** section and click **Edit**. From the **Authentication policy** dropdown menu, select **One factor access** and click **Save**.
```

Test the new configurations by recovering a password for a user of your sample app:

1. Start your app. On the Sign-In Widget, click the **Forgot password?** link.

1. Add the email address for your test user.

1. Click the **Send me an email** link.

1. Check your test user's email inbox and wait for the **Account password reset** email from your Okta dev org.

   >**Note:** Stay in the same browser window as the application sign-in tab.

1. In the body of the email, click the **Reset Password** link.

> 1. On the **Reset your Okta password** widget, create and verify a new password. Click **Reset Password**. Your test user is now signed in to your sample app with a new password.

Reviewer: is it a widget? Or is it a dialog? or a page?
```suggestion
1. On the **Reset your Okta password** widget, create and verify a new password. Click **Reset Password**. Your test user is now signed in to your sample app with a new password.
```

Author: @susanharper-okta It's a dialog that looks like our widget. It's probably considered part of the SIW, which is why I used the term. But not sure the best way to reference it. Or work around it?
